opal/mca/memory/linux/memory_linux_component.c

```diff
@@ -162,6 +187,59 @@ static int linux_register(void)
                                             OPAL_INFO_LVL_3,
                                             MCA_BASE_VAR_SCOPE_READONLY,
                                             &opal_memory_linux_disable);
+    if (0 > ret) {
+        return ret;
+    }
+
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
+    mca_memory_linux_component.use_memalign = 0;
+    ret = mca_base_component_var_register(&mca_memory_linux_component.super.memoryc_version,
+                                 "memalign",
+                                 "[64 | 32 | 0] - Enable (64bit or 32bit)/Disable(0) memory"
+                                 "alignment for all malloc calls (default: 0).",
+                                 MCA_BASE_VAR_TYPE_INT,
+                                 NULL,
+                                 0,
+                                 0,
+                                 OPAL_INFO_LVL_9,
+                                 MCA_BASE_VAR_SCOPE_READONLY,
+                                 &mca_memory_linux_component.use_memalign);
+    if (0 > ret) {
+        return ret;
+    }
+    (void) mca_base_var_register_synonym(ret, "ompi", "btl", "openib",
+                                         "memalign",
+                                         MCA_BASE_VAR_SYN_FLAG_DEPRECATED);
+
+    mca_memory_linux_component.memalign_threshold = 12288;
+    ret = mca_base_component_var_register(&mca_memory_linux_component.super.memoryc_version,
+                                 "memalign_threshold",
+                                 "Allocating memory more than memory_linux_memalign_threshold"
+                                 "bytes will automatically be algined to the value of memory_linux_memalign bytes."
+                                 "(default: 12288)",
+                                 MCA_BASE_VAR_TYPE_SIZE_T,
+                                 NULL,
+                                 0,
+                                 0,
+                                 OPAL_INFO_LVL_9,
+                                 MCA_BASE_VAR_SCOPE_READONLY,
+                                 &mca_memory_linux_component.memalign_threshold);
+    if (0 > ret) {
+        return ret;
+    }
+    (void) mca_base_var_register_synonym(ret, "ompi", "btl", "openib",
+                                         "memalign_threshold",
+                                         MCA_BASE_VAR_SYN_FLAG_DEPRECATED);
+
+    if (mca_memory_linux_component.use_memalign != 32
+        && mca_memory_linux_component.use_memalign != 64
+        && mca_memory_linux_component.use_memalign != 0){
+        opal_show_help("help-opal-memory-linux.txt", "invalid mca param value",
+                       true, "Wrong memalign parameter value. Allowed values: 64, 32, 0.",
+                       "memory_linux_memalign is reset to 32");
+        mca_memory_linux_component.use_memalign = 32;
+    }
```

Comment:
If you use the enumerator to specify the allowable values for the MCA param when creating it, you won't need this additional check because the MCA param (and MPI_T) infrastructure will automatically do this check for you.